version: '3.8'

services:
  db:
    image: postgres:latest
    container_name: twoge-database-server
    environment:
      POSTGRES_DB: ${DB_DATABASE}
      POSTGRES_USER: ${DB_USER}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
    ports:
      - "${DB_PORT}:5432"
    volumes:
      - db-data:/var/lib/postgresql/data
      - ./init_sql_scripts:/docker-entrypoint-initdb.d
    networks:
    - twoge-app-network
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "admin", "-d", "twoge_db"]
      interval: 10s
      timeout: 5s
      retries: 3
    env_file: .env

  app:
    build: .
    restart: always
    container_name: twoge-app    
    ports:
      - "${PORT_BACKEND}:8080"
    networks:
    - twoge-app-network
    environment:
      DATABASE_URL: postgres://${DB_USER}:${DB_PASSWORD}@${DB_HOST}:${DB_PORT}/${DB_DATABASE}
      PORT: ${PORT_BACKEND}
    depends_on:
      db:
        condition: service_healthy
    env_file: .env

networks:
  twoge-app-network:
    driver: bridge

volumes:
  db-data: